pkg/util/hostname.go

```diff
@@ -254,6 +249,13 @@ func GetHostnameData() (HostnameData, error) {
 			expErr.Set(err.Error())
 			hostnameErrors.Set("aws", expErr)
 		}
+
+		ec2Hostname, _, _, err := getValidEc2Hostname(getEC2Hostname)
+		if err == nil && hostName != ec2Hostname && ec2.IsWindowsDefaultHostname(originalHostname) {
```

Comment:
I think the condition here would be simpler with the following conditions:

1. `provider` is `aws`, and
2. `ec2.IsWindowsDefaultHostname(hostName)`, and
3. `!config.Datadog.GetBool("ec2_use_windows_prefix_detection")`

and by simpler I mean that you don't have to rely on additional temp variables and an additional call to `getValidEC2Hostname`. What do you think?

Reply:
If hostname is `EC2AMAZ-XXX` and `config.Datadog.GetBool("ec2_use_windows_prefix_detection")` is ` false`, wouldn't `ecs.IsECSInstance() || ec2.IsDefaultHostname(hostName)` be `false` and then `provider` never equals to `aws`.